spec/ndx-ophys-devices.extensions.yaml

```diff
@@ -1,8 +1,216 @@
+  neurodata_type_inc: Device
+  doc: Extends Device to hold metadata on the Excitation Source.
+  attributes:
+  - name: model
+    dtype: text
+    doc: Model of excitation source device.
+    required: false
+  - name: illumination_type
+    dtype: text
+    doc: 'Type of illumination. Suggested values: LED, LAser or more specifically
+      Gas, Metal vapor, Solid state, Dye, Semiconductor, Free electron, etc.'
+  - name: excitation_wavelength_in_nm
+    dtype: float
+    doc: Excitation wavelength of the stimulation light (nanometers).
+  - name: peak_power_in_W
+    dtype: float
+    doc: Incident power of stimulation device (in Watts).
+    required: false
+  - name: peak_pulse_energy_in_J
+    dtype: float
+    doc: If device is pulsed light source, pulse energy (in Joules).
+    required: false
+  - name: intensity_in_W_per_m2
+    dtype: float
+    doc: Intensity of the excitation in W/m^2, if known.
+    required: false
+  - name: pulse_rate_in_Hz
+    dtype: float
+    doc: If device is pulsed light source, pulse rate (in Hz) used for stimulation.
+    required: false
+  - name: exposure_time_in_s
+    dtype: float
+    doc: Exposure time of the sample (in sec).
+    required: false
+- neurodata_type_def: Photodetector
+  neurodata_type_inc: Device
+  doc: Extends Device to hold metadata on the Photodetector.
+  attributes:
+  - name: model
+    dtype: text
+    doc: Model of photodetector device.
+    required: false
+  - name: detector_type
+    dtype: text
+    doc: 'Technology used to detect the light. Suggested values: CCD, Intensified
+      CCD, Analog video, PMT, Photodiode, CMOS, EBCCD, FTIR.'
+  - name: detected_wavelength_in_nm
+    dtype: float
+    doc: Wavelength detected by photodetector.
+  - name: gain
+    dtype: float
+    doc: Gain on the photodetector.
+- neurodata_type_def: DichroicMirror
+  neurodata_type_inc: Device
+  doc: Extends Device to hold a Dichroic Mirror.
+  attributes:
+  - name: cut_on_wavelength_in_nm
+    dtype: float
+    doc: Wavelength at which the mirror starts to transmit light more than reflect.
+    required: false
+  - name: cut_off_wavelength_in_nm
+    dtype: float
+    doc: Wavelength at which transmission shifts back to reflection,for mirrors with
+      complex transmission spectra.
+    required: false
+  - name: reflection_bandwidth_in_nm
+    dtype: float
+    doc: The range of wavelengths that are primarily reflected.The start and end wavelengths
+      needs to be specified.
+    required: false
+  - name: transmission_bandwidth_in_nm
+    dtype: float
+    doc: The range of wavelengths that are primarily transmitted.The start and end
+      wavelengths needs to be specified.
+    required: false
+  - name: angle_of_incidence_in_degrees
+    dtype: float
+    doc: Intended angle at which light strikes the mirror.
+    required: false
+  - name: model
+    dtype: text
+    doc: Model of the dichroic mirror.
+    required: false
+- neurodata_type_def: BandOpticalFilter
```

Comment:
First time seeing 'band' tacked onto the front of 'optical filter' - is this common? Have you seen this referred to before? (I know for types like bandpass, sure, but I didn't know they were from a common parent of all 'band' filters)

Reply:
We decided to go for this in the ndx-fiber-photometry. Leaving you the discussions here: 
https://github.com/catalystneuro/ndx-fiber-photometry/pull/12

I think @pauladkisson proposed this naming to include in one device specs for both the band-pass and band-stop filters that only need the definition of a central wavelength and bandwidth.

Reply:
Edge filters is a term to combine short-pass and long-pass (see [here](https://www.edmundoptics.com/knowledge-center/application-notes/optics/optical-filters/)), so band filters seemed an appropriate name to combine band-pass and band-stop (aka notch) filters.

Reply:
Sure, I guess my qualm here is that is sounds a little weird

Would it make any more or less sense to say `OpticalBandFilter`?

Reply:
meh, probably not since that splits up its parent `OpticalFilter`

Reply:
By specifying a sub-type of the `Filter` property, not the sub-type of `Optical`

Reply:
hmm, I guess optical bandpass filters is more common than bandpass optical filter (according to quick google), so maybe thats the more intuitive name?

Reply:
Just to be sure: I should change to `OpticalBandpassFilter` and `OpticalEdgepassFilter`.

Reply:
Yeah I think `OpticalBandFilter` and `OpticalEdgeFilter`